image_transport/CMakeLists.txt

```diff
@@ -1,69 +1,123 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
+
 project(image_transport)
 
-find_package(catkin REQUIRED
-  COMPONENTS
-    message_filters
-    pluginlib
-    rosconsole
-    roscpp
-    roslib
-    sensor_msgs
-)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-find_package(Boost REQUIRED)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  DEPENDS message_filters pluginlib rosconsole roscpp roslib sensor_msgs
-)
+find_package(ament_cmake REQUIRED)
+
+find_package(message_filters REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(tinyxml_vendor REQUIRED)
+find_package(TinyXML REQUIRED)
 
-include_directories(include ${catkin_INCLUDE_DIRS})
+include_directories(include)
 
-# Build libimage_transport
+# Build image_transport library
 add_library(${PROJECT_NAME}
+  SHARED
```

Comment:
If there is a use case for this to be built statically (maybe there is not), we recommend using `ament_cmake_ros` to allow people to chose to build statically or dynamically via the `BUILD_SHARED_LIBS` CMake option